libsrc/interface/readuser.cpp

```diff
@@ -105,20 +110,47 @@ namespace netgen
                 while (1)
                   {
                     int pi, hi;
-                    Point<3> p;
+		    Point<3> p;
+		    string p1tmp, p2tmp, p3tmp;
 
                     in >> pi;
                     if (pi == -1)
                       break;
 		    
                     in >> hi >> hi >> hi;
-                    in >> p(0) >> p(1) >> p(2);
-
-                    mesh.AddPoint (p);
+		    if (DnotationSet == false) {
+			in >> p1tmp >> p2tmp >> p3tmp;
+			if (p1tmp.find("D") != std::string::npos){
+			    DnotationSet = true;
+			    Dnotation = true;
+			    cout << "Attention: in your UNV file, D is used as an exponent prefix instead of E" << endl;
+			    std::replace( p1tmp.begin(), p1tmp.end(), 'D', 'E');
+			    std::replace( p2tmp.begin(), p2tmp.end(), 'D', 'E');
+			    std::replace( p3tmp.begin(), p3tmp.end(), 'D', 'E');
+			}
+			p(0) = std::stod(p1tmp);
+			p(1) = std::stod(p2tmp);
+			p(2) = std::stod(p3tmp);
+		    }
+		    else if (Dnotation == true) {
+		        in >> p1tmp >> p2tmp >> p3tmp;
+			std::replace( p1tmp.begin(), p1tmp.end(), 'D', 'E');
+			std::replace( p2tmp.begin(), p2tmp.end(), 'D', 'E');
+			std::replace( p3tmp.begin(), p3tmp.end(), 'D', 'E');
+			p(0) = std::stod(p1tmp);
+			p(1) = std::stod(p2tmp);
+			p(2) = std::stod(p3tmp);
+		    }
+		    else{
+			in >> p(0) >> p(1) >> p(2);
+		    }
+		    mesh.AddPoint(p);
```

Comment:
Not sure how much this slows down parsing. If there is D notation, is it in every point? Then you could just check  the first one and then stay in one branch. 
If not you can consider using xbool -> a ngsolve class that can be true/false/maybe and as long as you didnt find E or D you can set it to maybe and then go to one branch.

Reply:
You are right, it is in every point. I changed the code so that it checks only the first line and then stays in one branch as you suggested. Thanks for pointing this out!